### PR TITLE
Reverts the service deletion logic

### PR DESCRIPTION
**Pull request by rewanthtammana**
This PR:

- removes the service deletion logic from crd-job
- removes the service deletion capability from crd-rbac

### Testing

Description on how trivy-operator can be tested.

- [ ] fresh install works
  - [ ] AWS
  - [ ] Azure
  - [ ] KVM
- [ ] upgrade from previous version works
  - [ ] AWS
  - [ ] Azure
  - [ ] KVM

#### Other testing

Description of features to additionally test for trivy-operator installations.

- [ ] check reconciliation of existing resources after upgrading
- [ ] X still works after upgrade
- [ ] Y is installed correctly

### Checklist

- [ ] Update changelog in CHANGELOG.md.
- [ ] Make sure `values.yaml` and `values.schema.json` are valid.

### CODE REVIEW

File: helm/trivy-operator/templates/crd-install/crd-job.yaml [hunk 1]
```diff
@@ -41,9 +41,6 @@ spec:
           # piping stderr to stdout means kubectl's errors are surfaced
           # in the pod's logs.
 
-          # Workaround for broken upgrade path going to 0.1.9: https://github.com/aquasecurity/trivy-operator/issues/314
-          kubectl delete service -n $pod_namespace trivy-operator || true
-
           kubectl apply -f /data/ 2>&1
         env:
```
Comment (fhielpos) [inline helm/trivy-operator/templates/crd-install/crd-job.yaml:45]:
```suggestion
```
I think this whole `env` section needs to go too

Comment (rewanthtammana) [reply helm/trivy-operator/templates/crd-install/crd-job.yaml]: Correct, done. I thought `pod_namespace` env might be used elsewhere, so I didn't remove it. I verified & it's not being used anywhere.

**Review by fhielpos — APPROVED** [review]

**Comment by fhielpos** [comment]
ℹ️ Remember to only **squash and merge**

**Comment by rewanthtammana** [comment]
Closes https://github.com/giantswarm/giantswarm/issues/24257. The code merged to main branch, yet to make a release.

**Pull request merged**